File: android/app/src/main/java/com/breathflow/app/java/exercise_coaching/ExerciseAnalyzerFactory.java
package com.breathflow.app.java.exercise_coaching;

import com.breathflow.app.java.exercise_coaching.exercises.BaseExerciseAnalyzer;
import com.breathflow.app.java.exercise_coaching.exercises.GenericExerciseAnalyzer;
import com.breathflow.app.java.exercise_coaching.exercises.NeckSideBendAnalyzer;

import java.util.List;
import java.util.Map;

/**
 * Egzersiz analizörleri için Factory sınıfı.
 * Yeni egzersiz analizörleri buraya eklenerek genişletilebilir.
 */
public class ExerciseAnalyzerFactory {

    /**
     * Egzersiz ID'sine göre uygun analizörü döndürür.
     */
    public static BaseExerciseAnalyzer createAnalyzer(String exerciseId) {
        if (exerciseId == null)
            return null;

        switch (exerciseId.toLowerCase()) {

            // ✅ Implement edilmiş egzersizler
            case "neck_side_bend":
            case "neck_lateral_flexion":
            case "neckmovement": // Flutter ExerciseConfig.neckMovement()
                return new NeckSideBendAnalyzer();

            // Diğer egzersizler için generic analyzer
            default:
                return new GenericExerciseAnalyzer(exerciseId, exerciseId, null, null);
        }
    }

    /**
     * Kural tabanlı generic analizör
     */
    public static BaseExerciseAnalyzer createAnalyzer(String analyzerType, Map<String, Object> rules,
            List<Map<String, Object>> feedbackRules) {
        String exerciseName = rules != null && rules.containsKey("exerciseName") ? (String) rules.get("exerciseName")
                : "Generic";
        String exerciseType = rules != null && rules.containsKey("exerciseType") ? (String) rules.get("exerciseType")
                : "generic";
        return new GenericExerciseAnalyzer(exerciseName, exerciseType, rules, feedbackRules);
    }

    /**
     * Desteklenen egzersiz ID'lerini döndürür
     */
    public static String[] getSupportedExerciseIds() {
        return new String[] {
                "neck_side_bend",
                "neck_lateral_flexion",
                "neckmovement"
        };
    }
}

File: ios/Runner/ExerciseCoaching/ExerciseAnalyzerFactory.swift
import Foundation

/// Egzersiz analizörleri için Factory sınıfı.
/// Android'deki ExerciseAnalyzerFactory.java karşılığı.
class ExerciseAnalyzerFactory {
    
    /// Egzersiz ID'sine göre uygun analizörü döndürür.
    static func createAnalyzer(exerciseId: String) -> ExerciseAnalyzer? {
        guard !exerciseId.isEmpty else { return nil }
        
        switch exerciseId.lowercased() {
        case "neck_side_bend", "neck_lateral_flexion", "neckmovement":
            return NeckSideBendAnalyzer()
        case "squat":
            return SquatAnalyzer()
        case "shoulder_stretch", "shoulderstretch":
            return ShoulderStretchAnalyzer()
        default:
            return GenericExerciseAnalyzer(exerciseName: exerciseId, exerciseType: exerciseId, rules: nil, feedbackRules: nil)
        }
    }
    
    /// Kural tabanlı generic analizör
    static func createAnalyzer(analyzerType: String, rules: [String: Any]?, feedbackRules: [[String: Any]]?) -> ExerciseAnalyzer {
        let exerciseName = rules?["exerciseName"] as? String ?? "Generic"
        let exerciseType = rules?["exerciseType"] as? String ?? "generic"
        return GenericExerciseAnalyzer(exerciseName: exerciseName, exerciseType: exerciseType, rules: rules, feedbackRules: feedbackRules)
    }
    
    /// Desteklenen egzersiz ID'leri
    static let supportedExerciseIds = [
        "neck_side_bend", "neck_lateral_flexion", "neckmovement",
        "squat",
        "shoulder_stretch", "shoulderstretch"
    ]
}
